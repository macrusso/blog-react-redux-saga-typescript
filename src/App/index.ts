import * as appConstants from "./appConstants";
import * as appActions from "./appActions";
import * as appSelectors from "./appSelectors";
import appReducer from "./appReducer";
import AppContainer from "./AppContainer";
import App from "./App";
import AppNavbar from "./AppNavbar";
import IAppState from "./IAppState";

export { App, AppNavbar, IAppState, appActions, appReducer, appSelectors, AppContainer, appConstants };
