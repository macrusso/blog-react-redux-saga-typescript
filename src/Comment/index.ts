
import * as commentConstants from './commentConstants';
import * as commentSelectors from './commentSelectors';
import * as commentActions from './commentActions';
// import CommentContainer from './CommentContainer';
import * as commentSagas from './commentSagas';
import commentReducer from './commentReducer';
import IComment from './IComment';

export {
  // Comment,
  IComment,
  commentSagas,
  commentActions,
  commentReducer,
  commentSelectors,
  commentConstants,
  // CommentContainer,
};