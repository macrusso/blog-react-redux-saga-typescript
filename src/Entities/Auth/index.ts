import * as authConstants from './authConstants';
import * as authActions from './authActions';
import * as authSagas from './authSagas';
import authReducer, { ICommentsState } from './authReducer';

export { authSagas, ICommentsState, authActions, authReducer, authConstants };
