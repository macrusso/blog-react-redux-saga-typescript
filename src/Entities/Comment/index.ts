import * as commentConstants from './commentConstants';
import * as commentSelectors from './commentSelectors';
import * as commentActions from './commentActions';
import * as commentSagas from './commentSagas';
import commentReducer, { ICommentsState } from './commentReducer';
import ICommentPartial from './ICommentPartial';
import IComment from './IComment';

export {
  IComment,
  commentSagas,
  ICommentsState,
  commentActions,
  commentReducer,
  ICommentPartial,
  commentSelectors,
  commentConstants,
};
