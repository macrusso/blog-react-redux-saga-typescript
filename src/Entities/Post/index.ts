import * as postConstants from "./postConstants";
import * as postSelectors from "./postSelectors";
import * as postActions from "./postActions";
import * as postSagas from "./postSagas";
import postReducer from "./postReducer";
import Post from "../../Frontend/Post/PostListItem";
import IPostPartial from "./IPostPartial";
import IPost from "./IPost";
import IPostsState from "./IPostsState";

export { Post, IPost, postSagas, postActions, IPostsState, postReducer, IPostPartial, postSelectors, postConstants };
