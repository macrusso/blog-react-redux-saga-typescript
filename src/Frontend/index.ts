export * from './Post';
export * from './Common';
export * from './PostAdd';
export * from './Comment';
export * from './CommentAdd';
