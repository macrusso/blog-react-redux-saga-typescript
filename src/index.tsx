import * as React from "react";
import * as ReactDOM from "react-dom";
import { Provider } from "react-redux";
import { applyMiddleware, compose, createStore, combineReducers } from "redux";
import { createBrowserHistory } from "history";
import { connectRouter, routerMiddleware } from "connected-react-router";
import createSagaMiddleware from "redux-saga";
import { AppContainer } from "./App";
import "./index.css";
import rootReducer from "./rootReducer";
import * as serviceWorker from './serviceWorker';
import rootSaga from "./rootSagas";

const basename = "/";
const history = createBrowserHistory({ basename });
const sagaMiddleware = createSagaMiddleware();
const middleware = applyMiddleware(routerMiddleware(history), sagaMiddleware);

/* eslint-disable no-underscore-dangle */
const composeEnhancers = (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

(window as any).__MUI_USE_NEXT_TYPOGRAPHY_VARIANTS__ = true;
/* eslint-enable */

const store = createStore(combineReducers({
  router: connectRouter(history),
  rootReducer,
}), composeEnhancers(middleware));

sagaMiddleware.run(rootSaga);

ReactDOM.render(
  <Provider store={store}>
    <AppContainer history={history} />
  </Provider>,
  document.getElementById("root") as HTMLElement,
);
serviceWorker.unregister();
