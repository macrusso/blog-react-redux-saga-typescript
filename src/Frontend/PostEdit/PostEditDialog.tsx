import React from 'react';
import {
  Dialog,
  DialogTitle,
  DialogContent,
  DialogActions,
  Button,
} from '@material-ui/core';
import { IPost } from 'src/Entities';

interface IPostEditDialog {
  post: IPost;
  open: boolean;
  handleCloseDialog: () => void;
  updatePost: (post: IPost) => void;
}

const PostEditDialog: React.SFC<IPostEditDialog> = props => {
  const { open, post, updatePost, handleCloseDialog } = props;

  return (
    <Dialog open={open} onClose={handleCloseDialog}>
      <DialogTitle>Edit your post</DialogTitle>
      <DialogContent>
        Let Google help apps determine location. This means sending anonymous
        location data to Google, even when no apps are running.
      </DialogContent>
      <DialogActions>
        <Button onClick={handleCloseDialog} color="primary" variant="outlined">
          Cancel
        </Button>
        <Button
          onClick={() => updatePost(post)}
          color="primary"
          variant="contained"
        >
          Submit
        </Button>
      </DialogActions>
    </Dialog>
  );
};

export default PostEditDialog;
