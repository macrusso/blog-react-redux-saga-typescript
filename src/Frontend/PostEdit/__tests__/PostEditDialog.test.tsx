import React from "react";
import { BrowserRouter } from "react-router-dom";
import { shallow } from "enzyme";
import PostEditDialog from "../PostEditDialog";
import { IPost } from "src/Entities";

describe("PostEditDialog", () => {
  test("Snapshot matches", () => {
    const post: IPost = {
      _id: "5bcc554259e0dbfda6ed1a55",
      userId: "5bcc554259e0dbfda6ed1a55",
      title: "string",
      body: "string",
    };

    const wrapper = shallow(
      <BrowserRouter>
        <PostEditDialog post={post} open={true} updatePost={jest.fn()} handleCloseDialog={jest.fn()} />
      </BrowserRouter>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
