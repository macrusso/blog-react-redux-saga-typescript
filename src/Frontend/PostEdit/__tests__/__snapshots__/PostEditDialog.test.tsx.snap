// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostEditDialog Snapshot matches 1`] = `ShallowWrapper {}`;
